src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkCapasity();
+        createAndInsertNewNode(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
+
+        Node<K, V> nextNode = table[key == null ? 0 : getBucketIndex(key.hashCode(), table.length)];
+        while (nextNode != null) {
+            if (nextNode.key == key
+                    || nextNode.key != null && nextNode.key.equals(key)) {
```

Comment:
as of now you have your own key comparison logic duplicated here and in `isNodeEqual()`